spec/features/course/experience_points_disbursement_spec.rb

```diff
@@ -17,9 +22,23 @@
     context 'As a Course Teaching Assistant' do
       let(:user) { course_teaching_assistant.user }
 
+      scenario 'I can filter students by group' do
+        approved_course_students
+        visit disburse_experience_points_course_users_path(course)
+        approved_course_students.each do |student|
+          expect(page).to have_content_tag_for(student)
+        end
+
+        click_link group.name
+        expect(page).not_to have_content_tag_for(ungrouped_course_student)
+        grouped_course_students.each do |student|
+          expect(page).to have_content_tag_for(student)
+        end
+      end
+
       scenario 'I can disburse experience points' do
         unapproved_course_student
-        unawarded_student, awarded_student, awarded_zero_student = course_students
+        unawarded_student, awarded_student, awarded_zero_student = approved_course_students
```

Comment:
Okay, understood - a sidenote: confusion probably came from the fact that the initial creation of objects does mix in some `group` objects (the alternative would be to create it within the scenario itself). Not sure if we need to change it now though.